pkg/importer/imageio-datasource.go

```diff
@@ -0,0 +1,489 @@
+		//Path provided is invalid.
+		return ProcessingPhaseError, ErrInvalidPath
+	}
+	file := filepath.Join(path, tempFile)
+	err := util.StreamDataToFile(is.readers.TopReader(), file)
+	if err != nil {
+		return ProcessingPhaseError, err
+	}
+	// If we successfully wrote to the file, then the parse will succeed.
+	is.url, _ = url.Parse(file)
+	return ProcessingPhaseProcess, nil
+}
+
+// TransferFile is called to transfer the data from the source to the passed in file.
+func (is *ImageioDataSource) TransferFile(fileName string) (ProcessingPhase, error) {
+	is.readers.StartProgressUpdate()
+	err := util.StreamDataToFile(is.readers.TopReader(), fileName)
+	if err != nil {
+		return ProcessingPhaseError, err
+	}
+	return ProcessingPhaseResize, nil
+}
+
+// Process is called to do any special processing before giving the URI to the data back to the processor
+func (is *ImageioDataSource) Process() (ProcessingPhase, error) {
+	return ProcessingPhaseConvert, nil
+}
+
+// GetURL returns the URI that the data processor can use when converting the data.
+func (is *ImageioDataSource) GetURL() *url.URL {
+	return is.url
+}
+
+// Close all readers.
+func (is *ImageioDataSource) Close() error {
+	var err error
+	if is.readers != nil {
+		err = is.readers.Close()
+	}
+	is.cancelLock.Lock()
+	if is.cancel != nil {
+		is.cancel()
+		is.cancel = nil
+	}
+	is.cancelLock.Unlock()
+	return err
+}
+
+func (is *ImageioDataSource) pollProgress(reader *util.CountingReader, idleTime, pollInterval time.Duration) {
+	count := reader.Current
+	lastUpdate := time.Now()
+	for {
+		if count < reader.Current {
+			// Some progress was made, reset now.
+			lastUpdate = time.Now()
+			count = reader.Current
+		}
+
+		if time.Until(lastUpdate.Add(idleTime)).Nanoseconds() < 0 {
+			is.cancelLock.Lock()
+			if is.cancel != nil {
+				// No progress for the idle time, cancel http client.
+				is.cancel() // This will trigger dp.ctx.Done()
+			}
+			is.cancelLock.Unlock()
+		}
+		select {
+		case <-time.After(pollInterval):
+			continue
+		case <-is.ctx.Done():
+			return // Don't leak, once the transfer is cancelled or completed this is called.
+		}
+	}
+}
+
+func createImageioReader(ctx context.Context, ep string, accessKey string, secKey string, certDir string, diskID string) (io.ReadCloser, uint64, error) {
+	conn, err := newOvirtClientFunc(ep, accessKey, secKey)
+	if err != nil {
+		return nil, uint64(0), errors.Wrap(err, "Error creating connection")
+	}
+	defer conn.Close()
+
+	it, total, err := getTransfer(conn, diskID)
+	if err != nil {
+		return nil, uint64(0), err
+	}
+
+	ca, err := loadCA(certDir)
+	if err != nil {
+		return nil, uint64(0), err
+	}
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			RootCAs: ca,
+		},
+	}
+	client := &http.Client{Transport: transport}
+	transferURL, available := it.TransferUrl()
+	if !available {
+		return nil, uint64(0), errors.New("Error transfer url not available")
+	}
+
+	signedTicket, available := it.SignedTicket()
```

Comment:
The daemon never used the signed ticket, and the proxy does not use it since 4.3.

It can be removed if you want to support oVirt >= 4.3.